extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/runtime/PanacheQueryImpl.java

```diff
@@ -100,11 +97,15 @@ public int pageCount() {
 
     @Override
     public Page page() {
-        checkNotInRange();
+        checkPagination();
         return page;
     }
 
-    private void checkNotInRange() {
+    private void checkPagination() {
+        if (page == null) {
+            throw new UnsupportedOperationException("Cannot call a page related method, " +
+                    "call page(Page) or page(int, int) to initiate pagination first");
+        }
```

Comment:
Well that's a change in behaviour: we used to have a single page encompassing all results. Perhaps it's fine, but at the very least, this has to be mentioned in the javadocs for the `PanacheQuery` interface.

Reply:
Javadoc updated, Noticed too that there is another interface `ReactivePanacheQuery` that has the same behaviour, so I updated it too so all three of them match.